### PR TITLE
OCPBUGS-41675: Fix IBU conformance baseline tests

### PR DESCRIPTION
**Pull request by mresvanis**
This change fixes the IBU conformance-baseline tests, by skipping tests that are expected to fail. The latter are related to the OADP and Lifecycle agent operator deployments.

The tests skipped for 4.1{6,7,8} `conformance-serial-baseline` are:

```
[sig-arch][Early] APIs for openshift.io must have stable versions
crd/dataprotectionapplications.oadp.openshift.io has an unstable version "v1alpha1" that is accessible-by-default. All CRDs accessible by default must be stable (v1, v2, etc) with guaranteed compatibility and upgradeability ~forever.
```
The tests skipped for 4.1{6,7,8} `conformance-parallel-baseline` are:
```
[sig-arch][Early] APIs for openshift.io must have stable versions
crd/dataprotectionapplications.oadp.openshift.io has an unstable version "v1alpha1" that is accessible-by-default. All CRDs accessible by default must be stable (v1, v2, etc) with guaranteed compatibility and upgradeability ~forever.
[sig-arch] Managed cluster should set requests but not limits
apps/v1/Deployment/openshift-adp/openshift-adp-controller-manager/container/manager defines a limit on cpu of 1 which is not allowed (rule: "apps/v1/Deployment/openshift-adp/openshift-adp-controller-manager/container/manager/limit[cpu]")
apps/v1/Deployment/openshift-adp/openshift-adp-controller-manager/container/manager defines a limit on memory of 512Mi which is not allowed (rule: "apps/v1/Deployment/openshift-adp/openshift-adp-controller-manager/container/manager/limit[memory]")
apps/v1/Deployment/openshift-lifecycle-agent/lifecycle-agent-controller-manager/container/kube-rbac-proxy does not have a cpu request (rule: "apps/v1/Deployment/openshift-lifecycle-agent/lifecycle-agent-controller-manager/container/kube-rbac-proxy/request[cpu]")
apps/v1/Deployment/openshift-lifecycle-agent/lifecycle-agent-controller-manager/container/kube-rbac-proxy does not have a memory request (rule: "apps/v1/Deployment/openshift-lifecycle-agent/lifecycle-agent-controller-manager/container/kube-rbac-proxy/request[memory]")
  
[sig-node][apigroup:config.openshift.io] CPU Partitioning cluster infrastructure should be configured correctly
projects [openshift-adp] do not contain the annotation map[workload.openshift.io/allowed:management]
 
[sig-node][apigroup:config.openshift.io] CPU Partitioning cluster platform workloads should be annotated correctly for Deployments
deployment (openshift-adp-controller-manager) in openshift namespace (openshift-adp) must have pod templates annotated with map[target.workload.openshift.io/management:{\"effect\": \"PreferredDuringScheduling\"}]
 
[sig-arch] Managed cluster should ensure platform components have system-* priority class associated
2 pods found with invalid priority class (should be openshift-user-critical or begin with system-):
openshift-adp/openshift-adp-controller-manager-799858d4f7-lbp2t (currently "")
openshift-lifecycle-agent/lifecycle-agent-controller-manager-7f4776f7c6-9nlnm (currently "")
```

### CODE REVIEW

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.15-upgrade-from-stable-4.14-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.15-upgrade-from-stable-4.14-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.16-upgrade-from-stable-4.15-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.16-upgrade-from-stable-4.15-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.18-upgrade-from-stable-4.17-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.18-upgrade-from-stable-4.17-image-based-upgrade-e2e-parallel-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse  periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline  periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.18-upgrade-from-stable-4.17-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.18-upgrade-from-stable-4.17-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.15-upgrade-from-stable-4.14-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.15-upgrade-from-stable-4.14-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.16-upgrade-from-stable-4.15-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.16-upgrade-from-stable-4.15-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.18-upgrade-from-stable-4.17-image-based-upgrade-e2e-serial-conformance-baseline periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.18-upgrade-from-stable-4.17-image-based-upgrade-e2e-parallel-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.15-upgrade-from-stable-4.14-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.15-upgrade-from-stable-4.14-image-based-upgrade-e2e-serial-conformance-baseline

**Comment by mresvanis** [comment]
/pj-rehearse periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.15-upgrade-from-stable-4.14-image-based-upgrade-e2e-serial-conformance-baseline

**Review by jeff-roche — APPROVED** [review]

**Comment by deads2k** [comment]
Can we get a complete list in a readable chart form?  the majority of these tests appear to be things that should **not** be skipped.

for instance, look at `Kubectl delete interactive based on user confirmation input`what about IBU breaks `kubectl` and why would we consider it acceptable to be unable to use the standard client against an IBU cluster.

This PR needs to be reverted and more carefully vetted prior to the next merge.

**Comment by deads2k** [comment]
quick summary from one stanza I can read

1. APIs for openshift.io must have stable versions - IBU must not add additional unstable APIs.
2. Kubectl delete interactive based on user confirmation input - IBU clusters must work with standard kubernetes tooling. kubectl is the CLI tool of record.
3. CPU Partitioning cluster platform workloads should be annotated correctly for Deployments - CPU partitioning works based on admission.  All admission plugins must continue to operate properly.
4. Managed cluster should set requests but not limits - Control plane must continue to operate as single node operates.  Functional requests and limits are essential for proper operation under load.
5. Managed cluster should ensure platform components have system-\* priority class associated - Control plane must continue to operate as single node operates.  Functional requests and limits are essential for proper operation under load.
6. CPU Partitioning cluster infrastructure should be configured correctly - I'm not immediately familiar with this one, but I doubt that it's an exception to "control plane must be functionally equivalent"
7. oc adm must-gather runs successfully - problem determination for CEE and engineering must remain functional
        for audit logs

**Comment by mresvanis** [comment]
> quick summary from one stanza I can read
> 
> 1. APIs for openshift.io must have stable versions - IBU must not add additional unstable APIs.
> 2. Kubectl delete interactive based on user confirmation input - IBU clusters must work with standard kubernetes tooling. kubectl is the CLI tool of record.
> 3. CPU Partitioning cluster platform workloads should be annotated correctly for Deployments - CPU partitioning works based on admission.  All admission plugins must continue to operate properly.
> 4. Managed cluster should set requests but not limits - Control plane must continue to operate as single node operates.  Functional requests and limits are essential for proper operation under load.
> 5. Managed cluster should ensure platform components have system-* priority class associated - Control plane must continue to operate as single node operates.  Functional requests and limits are essential for proper operation under load.
> 6. CPU Partitioning cluster infrastructure should be configured correctly - I'm not immediately familiar with this one, but I doubt that it's an exception to "control plane must be functionally equivalent"
> 7. oc adm must-gather runs successfully - problem determination for CEE and engineering must remain functional
>    for audit logs

@deads2k thank you for the feedback. 

TL;DR I will propose another change to remove the SKIP_TESTS and run conformance on SNO without non OCP workloads. 

The tests changed here refer to the conformance of SNO before IBU, so these are not tests failing on an IBU SNO. We skipped them temporarily to be able to work on the actual failures, i.e. conformance tests after IBU is performed on a SNO ([example](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-kni-lifecycle-agent-main-nightly-4.17-upgrade-from-stable-4.16-image-based-upgrade-e2e-parallel-conformance/1840396521737031680) of such failure). 

I agree that we shouldn't skip them, I'm working on running these tests before we add any non OCP workload (i.e. the OADP and the Lifecycle Agent operator, which lead to these failures we see here) to the SNO. The conformance of these workloads is not what we should be testing for, but OpenShift's conformance after an IBU is.

**Comment by mresvanis** [comment]
> > OADP and the Lifecycle Agent operator, which lead to these failures we see here
> 
> There are failures (kubectl and oc adm must-gather, for instance) that appear to be unrelated to these processes.

We were incorrectly using older versions of the clients (OCP 4.14 to be exact), instead of the versions that we should be using alongside each release version that we were testing on. I should have documented these in this PR's description though, so I apologize for not thoroughly explaining all the tests we skipped, why we did so and how we were going to alleviate those.
 
> 
> > OADP and the Lifecycle Agent operator, which lead to these failures we see here
> 
> How about updating the OADP and lifecycle agent to follow control plane best practices that will improve the availability and resiliency of these processes? For instance, adding priority classes and requests is pretty straight forward and will improve load shedding behavior. Similar for the CPU partitioning annotations.

I am checking in with the owners of those operators to see what's feasible, e.g. OADP using an unstable API is not something we can push forward at this time. I agree though that the rest of the requirements are feasible. I will keep you posted.
